sympy/stats/frv.py

```diff
@@ -375,12 +372,113 @@ class SingleFinitePSpace(SinglePSpace, FinitePSpace):
     def domain(self):
         return SingleFiniteDomain(self.symbol, self.distribution.set)
 
+    @property
+    def _is_symbolic(self):
+        """
+        Helper property to check if the distribution
+        of the random variable is having symbolic
+        dimension.
+        """
+        return self.distribution.is_symbolic
+
+    @property
+    def distribution(self):
+        return self.args[1]
+
+    def pmf(self, expr):
+        return self.distribution.pmf(expr)
+
     @property
     @cacheit
     def _density(self):
         return dict((FiniteSet((self.symbol, val)), prob)
                     for val, prob in self.distribution.dict.items())
 
+    @cacheit
+    def compute_characteristic_function(self, expr):
+        if self._is_symbolic:
+            d = self.compute_density(expr)
+            t = Dummy('t', real=True)
+            ki = Dummy('ki')
+            return Lambda(t, Sum(d(ki)*exp(I*ki*t), (ki, self.args[1].low, self.args[1].high)))
+        expr = rv_subs(expr, self.values)
+        return FinitePSpace(self.domain, self.distribution).compute_characteristic_function(expr)
```

Comment:
Maybe I misunderstand what this is doing but wouldn't the normal way to express this be
```
        return FinitePSpace.compute_characteristic_function(self, expr)
```
or something with `super`?

Since this happens in every method it seems like the `is_symbolic` attribute is actually emulating a subclass.